### ISSUE TITLE
False positive - eslint-plugin-react(jsx-key): Missing "key" prop for element in array. (again)

### ISSUE THREAD

**Issue by maurice** [oxc-project/oxc#1982]
I previously reported this in #1857 and I provided a fix with #1858 but clearly my fix wasn't good enough because I discovered a new and pretty similar issue with the latest `oxlint` binary.

Specifically the previous issue/fix worked for this kind of thing

```jsx
MyStory.decorators = [
  (Component) => (
    <div style={{ margin: "3em" }}>
      <Component />
    </div>
  ),
];
```

but not this

```jsx
MyStory.decorators = [
  (Component) => {
    const store = useMyStore();
    return (
      <Provider store={store}>
        <Component />
      </Provider>
    );
  }
];
```

I'm gonna try and make a PR for that too... 😄

**Comment by camc314** [comment]
Looks like there weren't enough test cases in eslint plugin react 🙂
